pkg/cfn/manager/api.go

```diff
@@ -450,15 +450,21 @@ func (c *StackCollection) DescribeStacks() ([]*Stack, error) {
 	return stacks, nil
 }
 
-func IsClusterStack(stacks []*Stack) bool {
+func (c *StackCollection) IsClusterStack() (bool, error) {
+	stacks, err := c.DescribeStacks()
+	if err != nil {
+		return false, err
+	}
 	for _, stack := range stacks {
-		for _, output := range stack.Outputs {
-			if *output.OutputKey == "ClusterStackName" {
-				return true
+		if *stack.StackName == c.makeClusterStackName() {
+			for _, tag := range stack.Tags {
+				if *tag.Key == api.ClusterNameTag && *tag.Value == c.spec.Metadata.Name {
```

Comment:
`matchesClusterName` from this package also has this logic:

```suggestion
				if matchesClusterName(*tag.Key, *tag.Value, spec.Metadata.Name) {
```